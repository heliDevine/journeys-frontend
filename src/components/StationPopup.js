import React from 'react';
import PropTypes from 'prop-types';

export const StationPopup = ({ selectedStation, onClose }) => {
  const totalDistance = selectedStation.totalJourneyDistanceFromStation;
  const displayDistance = isNaN(totalDistance)
    ? 'No total distance calculated'
    : `${(totalDistance / 1000).toFixed(1)} kilometers`;
  const emissionsSaved = isNaN(totalDistance)
    ? ''
    : `${((totalDistance / 1000) * 107.5).toFixed(0)} gCO2`;

  return (
    <div className="popup">
      <div className="popup-content">
        <h2>GREEN HELSINKI</h2>
        <p>{displayDistance}</p>
        <p>CO2 emissions saved {emissionsSaved}</p>
        <button className="close-button" onClick={onClose}>
          Close
        </button>
      </div>
    </div>
  );
};

export default StationPopup;

StationPopup.propTypes = {
  selectedStation: PropTypes.shape({
    totalJourneyDistanceFromStation: PropTypes.number.isRequired,
  }),
  onClose: PropTypes.func.isRequired,
};
