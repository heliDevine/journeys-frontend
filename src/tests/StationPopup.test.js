import React from 'react';
import { render } from '@testing-library/react';
import StationPopup from '.././components/StationPopup';

describe('StationPopup', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<StationPopup />);

    expect(asFragment(<StationPopup />)).toMatchSnapshot();
  });
});
