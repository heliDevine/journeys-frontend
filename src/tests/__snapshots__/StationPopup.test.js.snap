// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationPopup renders correctly 1`] = `
<DocumentFragment>
  <div
    class="popup-container"
  >
    <div
      class="popup-content"
    >
      <h3>
        Test station
      </h3>
      <p>
        In total 1000 journeys departed from Test station in May 2021. That's No total distance calculated cycled and  less CO2 emissions compared to driving same distance by an average car.
      </p>
      <button
        class="popup-close-button"
      >
        Close
      </button>
    </div>
  </div>
</DocumentFragment>
`;
